plasmapy/particles/particle_class.py

```diff
@@ -2458,3 +2460,122 @@ def molecule(
 >>> def is_electron(particle: ParticleLike):
 ...     return particle == Particle("e-")
 """
+
+
+class Photon(AbstractParticle):
+    def __init__(self, quantity: Quantity = None):
+        self._set_default_photon_properties()
+        if quantity and not self._validate_quantity(quantity):
+            raise ValueError(
+                "Specify non negative quantity with units of Energy, wavelength, frequency or momentum."
+            )
+
+        if quantity is None:
+            self.energy = None
+            self.frequency = None
+            self.momentum = None
+            self.wavelength = None
+        else:
+            if self._isMomentum(quantity):
+                # p = (h/λ), f = c/λ
+                self.momentum = quantity.to(u.kg * u.m / u.s)
+                self.wavelength = const.h / self.momentum
+                self.frequency = const.c / self.wavelength
+                self.energy = const.h * self.frequency
```

Comment:
The only thing remaining to do here is to improve the test coverage. https://github.com/PlasmaPy/PlasmaPy/pull/1641/files -> https://app.codecov.io/gh/PlasmaPy/PlasmaPy/compare/1641?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr%20comments&utm_term=PlasmaPy shows that this is the only untested block. Could you add a test for this variant of Photon creation?

Reply:
Thank you @StanczakDominik I have fixed it now along with few more additions please take a look :)